src/data/bedrock/ArmorTrimMaterialTypeIdMap.php

```diff
@@ -53,21 +53,16 @@ final class ArmorTrimMaterialTypeIdMap{
 	private array $materialToId = [];
 
 	public function __construct(){
-		foreach(Materials::getAll() as $material){
-			$this->register(match($material){
-				Materials::AMETHYST() => Ids::AMETHYST,
-				Materials::COPPER() => Ids::COPPER,
-				Materials::DIAMOND() => Ids::DIAMOND,
-				Materials::EMERALD() => Ids::EMERALD,
-				Materials::GOLD() => Ids::GOLD,
-				Materials::IRON() => Ids::IRON,
-				Materials::LAPIS() => Ids::LAPIS,
-				Materials::NETHERITE() => Ids::NETHERITE,
-				Materials::QUARTZ() => Ids::QUARTZ,
-				Materials::REDSTONE() => Ids::REDSTONE,
-				default => throw new AssumptionFailedError("Unhandled armor trim material type")
-			}, $material);
-		}
+		$this->register(Ids::AMETHYST, Materials::AMETHYST());
+		$this->register(Ids::COPPER, Materials::COPPER());
+		$this->register(Ids::DIAMOND, Materials::DIAMOND());
+		$this->register(Ids::EMERALD, Materials::EMERALD());
+		$this->register(Ids::GOLD, Materials::GOLD());
+		$this->register(Ids::IRON, Materials::IRON());
+		$this->register(Ids::LAPIS, Materials::LAPIS());
+		$this->register(Ids::NETHERITE, Materials::NETHERITE());
+		$this->register(Ids::QUARTZ, Materials::QUARTZ());
+		$this->register(Ids::REDSTONE, Materials::REDSTONE());
```

Comment:
Sorry to go back and forth on this but I changed my mind about the match. I realized it'll still be useful to detect errors even if PHPStan doesn't see them.